server/api/users.js

```diff
@@ -1,48 +1,74 @@
-const router = require('express').Router();
+const router = require("express").Router();
 const {
-  models: { User, Cart },
-} = require('../db');
+  models: { User, Cart, Order, Product },
+} = require("../db");
 module.exports = router;
 
-router.get('/', async (req, res, next) => {
+router.get("/", async (req, res, next) => {
   try {
     const users = await User.findAll({
-      attributes: ['id', 'username'],
+      attributes: ["id", "username"],
     });
     res.json(users);
   } catch (err) {
     next(err);
   }
 });
 
-router.put('/:id', async (req, res, next) => {
+router.put("/:id", async (req, res, next) => {
   try {
-    const currentUser = await User.findByPk(req.params.id);
-    const productInCart = await Cart.findAll({
-      where: {
-        userId: req.params.id,
-        productId: req.body.product.id,
-      },
+    const puppies = await User.findByPk(req.params.id, {
+      include: [
+        {
+          model: Order,
+          where: {
+            type: "active",
+          },
+          required: false,
+          include: [Product],
+        },
+      ],
     });
-    if (!productInCart.length) {
-      await currentUser.addProduct(req.body.product.id, {
-        through: { quantity: req.body.quantityChange },
-      });
+    console.log(`puppies`, puppies);
+    let currentOrder = {};
+    //TODO
+    //add logic to check for open/closed orders
+    if (puppies.orders.length) {
+      currentOrder = puppies.orders[0];
+      console.log(`currentOrder`, currentOrder);
     } else {
-      const newQuantity = productInCart[0].quantity + req.body.quantityChange;
-      await productInCart[0].update({
-        quantity: newQuantity,
-      });
+      currentOrder = await Order.create({ userId: puppies.id });
+      console.log(`currentOrdernew`, currentOrder);
+
+      // await currentOrder.setUser(puppies);
     }
+    // console.log("puppies", puppies.toJSON());
+    //console.log("puppies", puppies);
+    const message = await currentOrder.incrementProduct(req.body.product.id, 1);
   } catch (err) {
     console.log(err);
   }
 });
 
-router.get('/:id', async (req, res, next) => {
+router.get("/:id", async (req, res, next) => {
```

Comment:
@drewerd15 was not able to get hit this GET route in the app, I believe because of how the Routes are hooked up in our React components. We could choose to merge this up without that but it would be nice to get the full slice working prior to merge. Let me know if you had it working in the UI; I could be missing something

Reply:
I'm not sure it's working on my end